src/app/games/lasca/LascaMove.ts

```diff
@@ -0,0 +1,141 @@
+import { Coord, CoordFailure } from 'src/app/jscaip/Coord';
+import { Move } from 'src/app/jscaip/Move';
+import { ArrayUtils } from 'src/app/utils/ArrayUtils';
+import { assert } from 'src/app/utils/assert';
+import { MoveEncoder } from 'src/app/utils/Encoder';
+import { Localized } from 'src/app/utils/LocaleUtils';
+import { MGPFallible } from 'src/app/utils/MGPFallible';
+import { MGPOptional } from 'src/app/utils/MGPOptional';
+import { MGPSet } from 'src/app/utils/MGPSet';
+import { JSONObject, JSONValue, JSONValueWithoutArray, Utils } from 'src/app/utils/utils';
+import { LascaState } from './LascaState';
+
+export class LascaMoveFailure {
+
+    public static readonly CAPTURE_STEPS_MUST_BE_DOUBLE_DIAGONAL: Localized = () => $localize`Capture must be double diagonal steps!`;
+
+    public static readonly MOVE_STEPS_MUST_BE_SINGLE_DIAGONAL: Localized = () => $localize`Move must be single diagonal steps!`;
```

Comment:
not user friendly !

Reply:
diagonal: pas enlevable, je sais pas être plus pédagogique que "diagonale" sans détester mon site :D
single step: pareil :D
HALP :D Je sais pas quoi faire :D

Reply:
c'est pas tellement sur diagonal ou step ma remarque, mais sur le phrasage: "Mouvement doit être simple pas diagonal!" -> "You must move in a single diagonal step"

Reply:
"Les déplacements doivent être des simple pas en diagonale !" et le FR est bon ?